src/test/java/com/rybka/repository/CurrencyHistoryRepositoryTest.java

```diff
@@ -57,17 +55,41 @@ public void testOnCorrectHistoryFindTop5() {
         var historyOtherList = List.of(
```

Comment:
I'd do in such way:

1. Create a new builder method with timestamp parameter.
2. Construct 5 history objects with date in the past (like 2020-07-09T22:00) //noiseList
3. Construct 5 history objects with current date(this will be our expected objects) //expectedList
4. Save to the DB
5. result = findTop5ByOrderByDateDesc //method will sort and retrieve the latest objects.
6. compare results.

Reply:
fixed